src/providers/python.rs

```diff
@@ -0,0 +1,134 @@
+use std::collections::HashMap;
+
+use anyhow::{Result, Context};
+use serde::Deserialize;
+
+use crate::{
+    nixpacks::{app::App, environment::Environment, nix::NixConfig},
+    Pkg, chain,
+};
+
+use super::Provider;
+
+pub struct PythonProvider {}
+impl Provider for PythonProvider {
+    fn name(&self) -> &str {
+        "python"
+    }
+
+    fn detect(&self, app: &crate::nixpacks::app::App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("main.py")
+            || app.includes_file("requirements.txt")
+            || app.includes_file("pyproject.toml"))
+    }
+
+    fn install_cmd(&self, app: &App, _env: &Environment) -> Result<Option<String>> {
+        if app.includes_file("requirements.txt") {
+            return Ok(Some(
+                "python -m ensurepip && python -m pip install -r requirements.txt".to_string(),
+            ));
+        } else if app.includes_file("pyproject.toml") {
+            return Ok(Some("python -m ensurepip && python -m pip install --upgrade build setuptools && python -m pip install .".to_string()));
+        }
+        Ok(None)
+    }
+
+    fn suggested_build_cmd(&self, _app: &App, _env: &Environment) -> Result<Option<String>> {
+        Ok(None)
+    }
+
+    fn suggested_start_command(&self, app: &App, _env: &Environment) -> Result<Option<String>> {
+        if app.includes_file("pyproject.toml") {
+            if let Ok(meta) = self.parse_pyproject(app) {
+                if let Some(entry_point) = meta.entry_point {
+                    return match entry_point {
+                        EntryPoint::Command(cmd) => Ok(Some(cmd)),
+                        EntryPoint::Module(module) => {
+                            Ok(Some(format!("python -m {}", module)))
```

Comment:
I think these should `return`
```suggestion
                        EntryPoint::Command(cmd) => return Ok(Some(cmd)),
                        EntryPoint::Module(module) => {
                            return Ok(Some(format!("python -m {}", module)))
```

Reply:
It's already returning the result of the `match` expression, so I'm not sure that that's really necessary.

Reply:
Oh nice, I missed that